pkg/azureclients/armclient/azure_armclient.go

```diff
@@ -192,10 +192,11 @@ func DoHackRegionalRetryDecorator(c *Client) autorest.SendDecorator {
 }
 
 // Send sends a http request to ARM service with possible retry to regional ARM endpoint.
-func (c *Client) Send(ctx context.Context, request *http.Request) (*http.Response, *retry.Error) {
+func (c *Client) Send(ctx context.Context, request *http.Request, decorators ...autorest.SendDecorator) (*http.Response, *retry.Error) {
```

Comment:
Is this change affecting anything? It seems there's no usage of `Send()` with `decorators`?

BTW, I feel like `Send()` should not be directly used by those azure clients but now there're many such usage... Do you think so?

Reply:
I'm trying to refactor ratelimiter and metrics collector. With this parameter, it is much easier to add log utils and tracing utils. 
And I think Send should be used by client because we can add customized wrapper for each request while all of the client still share the same http connections. Do you think so?

Reply:
Yes, it is for customized request. Hopefully I can find a way to improve but it may be hard.